src/main/java/lotto/domain/Ranking.java

```diff
@@ -0,0 +1,51 @@
+package lotto.domain;
+
+import java.util.Arrays;
+
+public enum Ranking {
+
+    FIRST(2000000000, 6, 0),
+    SECOND(30000000, 5, 1),
+    THIRD(1500000, 5, 0),
+    FOURTH(50000, 4, 0),
+    FIFTH(5000, 3, 0),
+    FAIL(0, 0, 0);
+
+    private final int winnerPrice;
+    private final int normalSuccessNumber;
+    private final int bonusSuccessNumber;
+
+    Ranking(int winnerPrice, int normalSuccessNumber, int bonusSuccessNumber) {
+        this.winnerPrice = winnerPrice;
+        this.normalSuccessNumber = normalSuccessNumber;
+        this.bonusSuccessNumber = bonusSuccessNumber;
+    }
+
+    public static Ranking judgeRanking(LottoResult lottoResult) {
+        return Arrays.stream(Ranking.values())
+            .filter(rank -> rank.equals(lottoResult))
+            .findFirst()
+            .orElse(FAIL);
+    }
+
+    private boolean equals(LottoResult lottoResult) {
+        return this.normalSuccessNumber == lottoResult.getNormalSuccessCount() &&
+            this.bonusSuccessNumber == lottoResult.getBonusSuccessCount();
+    }
+
+    public int multiplyCountAndWinnerPrice(final int count) {
+        return winnerPrice * count;
+    }
+
+    public boolean isFail() {
+        return this.equals(FAIL);
```

Comment:
enum 클래스는 항상 같은 인스턴스를 반환하기 때문에 같은 주소 값을 가지게 됩니다. 😃 
따라서 `equals`로 비교할 필요 없이 `==`의 동등성을 비교해도 같은 결과를 얻을 수 있어요! 😉

아래와 같이 변경할 수 있을 것 같아요! 🤗
```suggestion
        return this == FAIL;
```

Reply:
헉 그러네요!! 꼼꼼한 피드백 좋아요 😃